subsys/nfc/tnep/ch/common.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef NFC_TNEP_CH_COMMON_H_
+#define NFC_TNEP_CH_COMMON_H_
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include <net/buf.h>
+
+#include <nfc/ndef/ch.h>
+#include <nfc/tnep/ch.h>
+
+#ifdef _cplusplus
+extern "C" {
+#endif
+
+
+/* @brief Parse the Connection Handover Request Message.
+ *
+ * @param[in] msg Pointer to the Connection Request Message.
+ * @param[out] ch_req Pointer to the Connection Request structure.
+ * @param[in] buf Buffer for parsed data.
+ *
+ * @retval 0 If the operation was successful.
+ *         Otherwise, a (negative) error code is returned.
+ */
+int nfc_tnep_ch_request_msg_parse(const struct nfc_ndef_msg_desc *msg,
+				  struct nfc_tnep_ch_request *ch_req,
+				  struct net_buf_simple *buf);
+
+/* @brief Parse the Connection Handover Select Message.
+ *
+ * @param[in] msg Pointer to the Connection Select Message.
+ * @param[out] ch_select Pointer to the Connection Handover structure.
+ * @param[in] buf Buffer for parsed data.
+ *
+ * @retval 0 If the operation was successful.
+ *         Otherwise, a (negative) error code is returned.
+ */
+int nfc_tnep_ch_select_msg_parse(const struct nfc_ndef_msg_desc *msg,
+				 struct nfc_tnep_ch_record *ch_select,
+				 struct net_buf_simple *buf);
+
+/* @brief Parse the Connection Handover Initiate Message.
+ *
+ * @param[in] msg Pointer to the Connection Initiate Message.
+ * @param[out] ch_select Pointer to the Connection Handover structure.
+ * @param[in] buf Buffer for parsed data.
+ *
+ * @retval 0 If the operation was successful.
+ *         Otherwise, a (negative) error code is returned.
+ */
+int nfc_tnep_ch_initiate_msg_parse(const struct nfc_ndef_msg_desc *msg,
+				   struct nfc_tnep_ch_record *ch_init,
+				   struct net_buf_simple *buf);
+
+/* @brief Parse the Connection Handover Carrier Record.
+ *
+ * @param[in,out] hc_rec Connection Handover Carrier Record structure.
+ * @param[in] rec Pointer to the Handover Carrier Record descriptor.
+ * @param[in] buf Buffer for parsed data.
+ *
+ * @retval 0 If the operation was successful.
+ *         Otherwise, a (negative) error code is returned.
+ */
+int nfc_tnep_ch_hc_rec_parse(const struct nfc_ndef_ch_hc_rec **hc_rec,
+			     const struct nfc_ndef_record_desc *rec,
+			     struct net_buf_simple *buf);
```

Comment:
This is a set of the function which helps to parse all Connection Handover Record which can be included in single a little complicated NDEF message. At this moment use case for it is only the TNEP and in our case there won't be a other. If necessary it could be easily moved to NDEF section.